pkg/cri/server/images/snapshots.go

```diff
@@ -70,44 +70,55 @@ func (s *snapshotsSyncer) start() {
 func (s *snapshotsSyncer) sync() error {
 	ctx := ctrdutil.NamespacedContext()
 	start := time.Now().UnixNano()
-	var snapshots []snapshot.Info
-	// Do not call `Usage` directly in collect function, because
-	// `Usage` takes time, we don't want `Walk` to hold read lock
-	// of snapshot metadata store for too long time.
-	// TODO(random-liu): Set timeout for the following 2 contexts.
-	if err := s.snapshotter.Walk(ctx, func(ctx context.Context, info snapshot.Info) error {
-		snapshots = append(snapshots, info)
-		return nil
-	}); err != nil {
-		return fmt.Errorf("walk all snapshots failed: %w", err)
-	}
-	for _, info := range snapshots {
-		sn, err := s.store.Get(info.Name)
-		if err == nil {
-			// Only update timestamp for non-active snapshot.
-			if sn.Kind == info.Kind && sn.Kind != snapshot.KindActive {
-				sn.Timestamp = time.Now().UnixNano()
-				s.store.Add(sn)
-				continue
-			}
-		}
-		// Get newest stats if the snapshot is new or active.
-		sn = snapshotstore.Snapshot{
-			Key:       info.Name,
-			Kind:      info.Kind,
-			Timestamp: time.Now().UnixNano(),
+
+	for key, snapshotter := range s.snapshotters {
+		var snapshots []snapshot.Info
+		// Do not call `Usage` directly in collect function, because
+		// `Usage` takes time, we don't want `Walk` to hold read lock
+		// of snapshot metadata store for too long time.
+		// TODO(random-liu): Set timeout for the following 2 contexts.
+		if err := snapshotter.Walk(ctx, func(ctx context.Context, info snapshot.Info) error {
+			snapshots = append(snapshots, info)
+			return nil
+		}); err != nil {
+			return fmt.Errorf("walk all snapshots failed: %w", err)
```

Comment:
error messages should contain snapshotter names (e.g. `WithField("snapshotter", key)`)

Reply:
Fixed by adding `snapshotter` to the error message.